core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -284,26 +284,35 @@ class BrokerTopicMetrics(name: Option[String], remoteStorageEnabled: Boolean = f
   }
```

Comment:
@stanislavkozlovski @ijuma Sorry for pinging you on this PR, but I have a question about this sync of `MeterWrapper` (related to #7198)

This PR tries to align the behavior of new wrapper with `MeterWrapper`, and so I noticed the lock used by `MeterWrapper`. `MetricsRegistry` is a global variable in our code base, and both methods `MetricsRegistry#newXXX` and `MetricsRegistry#removeMetric` should be thread-safe. That means `newMeter` does NOT return `null` (the impl leverage `ConcurrentHashMap` to avoid race condition)

Hence, my question is "why we add extra lock to MeterWrapper"? Please correct me if I misunderstand anything. Thanks!